com.unity.multiplayer.mlapi/Prototyping/NetworkTransform.cs

```diff
@@ -297,34 +197,33 @@ private void FixedUpdate()
                 return;
             }
 
-            if (CanUpdateTransform())
+            if (IsNetworkStateDirty)
             {
-                m_NetworkPosition.Value = m_CurrentPosition;
-                m_NetworkRotation.Value = m_CurrentRotation;
-                m_NetworkWorldScale.Value = m_CurrentScale;
+                if (CanUpdateTransform)
+                {
+                    UpdateNetworkState();
+                }
+                else
+                {
+                    ApplyNetworkState(m_NetworkState.Value);
```

Comment:
Why do we need this? Isn't this already handled by OnNetworkStateChanged? Does this work with client authoritative players? (if you test with the player cube in testproject, does it move correctly for both client and host?)
I didn't put my state update in FixedUpdate, only on state change since my client auth cube's position would keep be overriden by the server

Reply:
huh, I tested this on `SampleScene` with host + client and everything worked perfectly! including different authority sphere pickups etc too. having said that, I'll give it another go and make sure I'm not skipping anything there. I'll follow-up soon.

Reply:
ah, I think I just left this here (wouldn't harm though) and yeah, not really needed :)

Reply:
because if a non-auth makes a move, this call would make it move back to the net-sync'd position.